### ISSUE TITLE
Values prop for PropertyOrBuilder is undefined with edit permissions false

### ISSUE THREAD

**Issue by faizanabidnaqvi** [Camberi/firecms#304]
Hi there!

I am working on a schema that is supposed to be read-only. Therefore, I set the permissions for the collection to be edit: false and create: false.

In the schema, I have some properties which use the PropertyBuilder so they can use values from other fields. 

However, values for my properties appeared undefined even when those values do exist and show up for the properties not using the builder. When I set the edit permissions to true, then the values object contains the entity's data.

I am using the latest release version.

Let me know if you have any follow-up questions. :)

**Comment by fgatti675** [comment]
I was not able to reproduce this bug!
Keep in mind that your PropertyBuilder may receive empty values.
What you return when receiving an empty value is the default property, for example the one defining the column props.

**Comment by faizanabidnaqvi** [comment]
Hey @fgatti675 

Thanks for getting back. I tried this again in the example project on the test entity and it does still happen.

Here is the schema where I am also logging the received values:
```
export const testEntitySchema = buildSchema({
    customId: true,
    name: "Test entity",
    properties: {
        
        source: ({ values, previousValues }) => {
            console.log(values)
            const properties = buildProperties<any>({
                type: {
                    dataType: "string",
                    config: {
                        enumValues: {
                            "facebook": "FacebookId",
                            "apple": "Apple"
                        }
                    }
                }
            });

            if (values.source) {
                if ((values.source as any).type === "facebook") {
                    properties["facebookId"] = buildProperty({
                        title: "Facebook id",
                        dataType: "string"
                    });
                } else if ((values.source as any).type === "apple") {
                    properties["appleId"] = buildProperty({
                        title: "Apple id",
                        dataType: "number"
                    });
                }
            }

            return ({
                dataType: "map",
                title: "Source",
                properties: properties
            });
        },
    },
    defaultValues: {
        // empty_string: "",
        title: null
    }
});
```

And collection:
```
export const testCollection = buildCollection({
    path: "test_entity",
    schema: testEntitySchema,
    name: "Test entity",
    group: "Example",
    permissions: ({ authController }) => ({
        edit: false,
        create: false,
    }),
});
```

With edit set to false, the console shows default values for the selected item (when opening side dialog) even though it has already been saved. The values do show up for the collection view but not in the side dialog.
<img width="1344" alt="Screenshot 2022-05-04 at 6 05 44 PM" src="https://user-images.githubusercontent.com/5749036/166723877-000c8a3d-fa9f-469d-ad0c-6209cd8c2972.png">

When edit is true, the console does show the values when side dialog is opened:
<img width="1366" alt="Screenshot 2022-05-04 at 6 06 20 PM" src="https://user-images.githubusercontent.com/5749036/166723996-c7ad3daf-4ef1-4f78-9f53-d6c0c31402c6.png">

Let me know if you need any further info!

**Comment by fgatti675** [comment]
Fixed in version 1.0.0 ;)
